cfme/tests/infrastructure/test_infra_quota_via_services.py

```diff
@@ -0,0 +1,141 @@
+        namespaces.instantiate('Cloud'). \
+        namespaces.instantiate('VM'). \
+        namespaces.instantiate('Provisioning'). \
+        namespaces.instantiate('StateMachines'). \
+        classes.instantiate('ProvisionRequestApproval'). \
+        instances.instantiate('Default')
+
+    yield instance
+    instance.delete()
+
+
+@pytest.fixture(scope='function')
+def max_field_method(request, max_quota_test_instance):
+    field, value = request.param
+    with update(max_quota_test_instance):
+        max_quota_test_instance.fields = {field: {'value': value}}
+    yield
+    with update(max_quota_test_instance):
+        max_quota_test_instance.fields = {field: {'value': ''}}
+
+
+def create_dialog(appliance, element_data, label=None):
+    if label is None:
+        label = 'label_' + fauxfactory.gen_alphanumeric()
+    sd = appliance.collections.service_dialogs.create(label=label,
+        description="my dialog", submit=True, cancel=True)
+    tab = sd.tabs.create(tab_label='tab_' + fauxfactory.gen_alphanumeric(),
+        tab_desc="my tab desc")
+    box = tab.boxes.create(box_label='box_' + fauxfactory.gen_alphanumeric(),
+        box_desc="my box desc")
+    box.elements.create(element_data=[element_data])
+    return sd
+
+
+@pytest.fixture(scope='module')
+def dialog_with_field(request, appliance):
+    field = request.param
+    element_data = {
+        'ele_label': "ele_" + fauxfactory.gen_alphanumeric(),
+        'ele_name': field,
+        'ele_desc': fauxfactory.gen_alphanumeric(),
+        'choose_type': "Text Box",
+        'default_text_box': "Default text"
+    }
+    dialog = create_dialog(appliance, element_data)
+    yield dialog
+    dialog.delete()
+
+
+def vmware_catalog_item(provider, vm_name, template, catalog, number_of_vms_dialog, prov_data):
+    prov_data['environment'] = {'automatic_placement': True}
+    catalog_item = CatalogItem(item_type='VMware',
+                               name=fauxfactory.gen_alphanumeric(),
+                               description="my catalog",
+                               display_in=True,
+                               provider=provider,
+                               catalog=catalog,
+                               catalog_name=template,
+                               dialog=number_of_vms_dialog,
+                               prov_data=prov_data)
+    catalog_item.create()
+    return catalog_item
+
+
+@pytest.mark.meta(blockers=[BZ(1497912, forced_streams=['5.7', '5.8', '5.9', 'upstream'])])
+@pytest.mark.parametrize(
+    ['max_field_method', 'dialog_with_field', 'prov_data'],
+    [
+        [('max_vms', '1'), 'number_of_vms', {'catalog': {'vm_name': vm_name, 'num_vms': '2'}}],
+        [('max_cpus', '1'), 'number_of_sockets', {'catalog': {'vm_name': vm_name},
+                                                  'hardware': {'num_sockets': '2'}}]
+    ],
+    indirect=['max_field_method', 'dialog_with_field'],
+    ids=['max_vms', 'max_cpus'])
```

Comment:
**REQUIRED**
It looks scary, can you somehow simplify it?

Reply:
Can you explain what scared you?